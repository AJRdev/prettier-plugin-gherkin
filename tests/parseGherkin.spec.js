const fs = require("fs");
const path = require("path");
const parseGherkin = require("./../src/parseGherkin");

describe("parseGherkin", () => {
  it("parse basic feature file into an AST", () => {
    const basicFeature = fs.readFileSync(
      path.join(__dirname, "fixtures/basic.feature"),
      { encoding: "utf-8" },
    );

    const ast = parseGherkin(basicFeature);

    expect(ast).toMatchSnapshot();
  });
});
