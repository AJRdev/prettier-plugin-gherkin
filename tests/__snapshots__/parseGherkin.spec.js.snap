// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`parseGherkin parse basic feature file into an AST 1`] = `
Object {
  "feature": Object {
    "children": Array [
      Object {
        "scenario": Object {
          "keyword": "Scenario",
          "location": Object {
            "column": 5,
            "line": 2,
          },
          "name": "minimalistic",
          "steps": Array [
            Object {
              "keyword": "Given ",
              "location": Object {
                "column": 9,
                "line": 3,
              },
              "text": "the minimalism",
            },
            Object {
              "keyword": "When ",
              "location": Object {
                "column": 9,
                "line": 4,
              },
              "text": "you change nothin",
            },
            Object {
              "keyword": "Then ",
              "location": Object {
                "column": 9,
                "line": 5,
              },
              "text": "it's still small",
            },
          ],
        },
      },
    ],
    "keyword": "Feature",
    "language": "en",
    "location": Object {
      "column": 1,
      "line": 1,
    },
    "name": "Minimal",
  },
}
`;
